uaclient/live_tree_model.py

```diff
@@ -0,0 +1,277 @@
+import logging
+
+from PyQt5.QtCore import (
+    pyqtSignal,
+    Qt,
+    QObject,
+    QRunnable,
+    QThreadPool,
+)
+from PyQt5.QtGui import QStandardItem, QIcon
+
+from asyncua import ua
+from asyncua.sync import SyncNode, new_node
+from asyncua.common.ua_utils import data_type_to_string
+
+from uawidgets.tree_widget import TreeViewModel
+
+
+logger = logging.getLogger(__name__)
+
+
+class DescriptionDataTypeLoader(QObject):
+    def update_description_and_data_type(self, node, description, data_type):
+        attrs = node.read_attributes(
+            [ua.AttributeIds.Description, ua.AttributeIds.DataType]
+        )
+        description.setText(attrs[0].Value.Value.Text)
+        if attrs[1].Value.Value:
+            data_type.setText(data_type_to_string(attrs[1].Value.Value))
+
+
+class NodeValueItem(QStandardItem):
+    def __init__(self, value):
+        super(NodeValueItem, self).__init__(value)
+
+    def update_value(self, node, val, data):
+        self.setText(str(val))
+
+
+class NodeSubscriptionSignal(QObject):
+    signal = pyqtSignal(object, object, str)
+
+    def __init__(self):
+        QObject.__init__(self)
+
+
+class SubscriptionHandler(QObject):
+    unsubscribe_started = pyqtSignal()
+    unsubscribe_ended = pyqtSignal()
+    subscribe_started = pyqtSignal()
+    subscribe_ended = pyqtSignal()
+
+    def __init__(self, model, idx):
+        QObject.__init__(self)
+        self.model = model
+        self.idx = idx
+        self.is_subscribe_running = False
+        self.is_unsubscribe_running = False
+
+        self.is_expanded = False
+        self.unsubscribe_is_waiting = False
+        self.subscribe_is_waiting = False
```

Comment:
Several of these appear to be used within the workers, which are running in other threads. I doubt their access is atomic; should we have mutexes here?